DFC.Digital/DFC.Digital.Repository.SitefinityCMS/Modules/JobProfileRepository.cs

```diff
@@ -41,11 +41,10 @@ public JobProfile GetByUrlName(string urlName)
             var key = urlName.ToLower();
             if (!cachedJobProfiles.ContainsKey(key))
             {
-                var jobProfile = ConvertDynamicContent(repository.Get(item => item.UrlName == urlName && item.Status == ContentLifecycleStatus.Live && item.Visible == true));
-                if (jobProfile?.IsImported != true)
-                {
-                    cachedJobProfiles.Add(key, jobProfile);
-                }
+                var jobProfile = ConvertDynamicContent(repository.Get(item =>
+                    item.UrlName == urlName && item.Status == ContentLifecycleStatus.Live && item.Visible == true));
+
+                cachedJobProfiles.Add(key, jobProfile?.IsImported == true ? null : jobProfile);
```

Comment:
Why are we adding null to cached collection?

Reply:
We are already implicitly adding null values to the collection when a job does not exist, this is just re written to ensure we have a value added to the collection on any urlname we search for.